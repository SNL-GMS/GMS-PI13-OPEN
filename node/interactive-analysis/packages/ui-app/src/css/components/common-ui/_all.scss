@import 'system-message';
@import 'base-display';
@import 'waveform-display';
@import 'toolbar';
@import 'sound-configuration';
