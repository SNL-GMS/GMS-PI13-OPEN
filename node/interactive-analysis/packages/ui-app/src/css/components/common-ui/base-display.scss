.base-display {
  display: flex;
  position: relative;
  flex-direction: column;
  &.full-width-height {
    height: 100%;
    width: 100%;
    flex-grow: 1;
  }
}
