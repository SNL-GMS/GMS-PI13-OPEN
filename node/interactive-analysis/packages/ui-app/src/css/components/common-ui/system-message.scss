$critical-selector: '[data-severity*=CRITICAL]';
$warning-selector: '[data-severity*=WARNING]';
$info-selector: '[data-severity*=INFO]';

@mixin createColoredCount($selector, $color) {
  #{$selector} &__count {
    color: lighten($color, 10%);
  }
}

.system-message-display {
  @include flexColumn;
  width: 100%;
  height: 100%;
  .toolbar {
    margin-bottom: 0.5em;
  }
}

.system-message-table {
  @include flexColumn;
  width: 100%;
  height: 100%;
  background-color: $gms-prominent-background;
  .system-cell {
    &#{$critical-selector} {
      background-color: $system-message-critical-color;
    }
    &#{$warning-selector} {
      background-color: $system-message-warning-color;
    }
    &#{$info-selector} {
      background-color: $system-message-info-color;
    }
  }
  &__button {
    &--floating {
      position: absolute;
      left: 50%;
      bottom: 3rem;
      transform: translate3D(-50%, -50%, 0);
    }
    &--desc {
      bottom: auto;
      top: 9rem;
    }
  }
}

.system-message-summary {
  @include centerChildren;
  background: $gms-prominent-background;
  height: 100%;
  padding: 0 1em;
  border-radius: $rounded-rectangle-radius;
  font-size: 1rem;

  @include createColoredCount($critical-selector, $gms-strong-warning);
  @include createColoredCount($warning-selector, $gms-warning);
  @include createColoredCount($info-selector, $gms-good-count);

  &__entry {
    @include centerChildren('vertical');
    cursor: pointer;
    & + & {
      margin-left: 1.5em;
    }
    &--disabled .system-message-summary__count {
      color: $gms-soft;
    }
  }

  &__count {
    font-family: $gms-mono;
    font-size: 1.5rem;
    font-weight: bold;
    margin-left: 0.5ch;
    &--prefix {
      opacity: 0.33;
    }
    &--main {
      opacity: 1;
    }
  }
}

/// AG Grid Overrides
.system-message-table {
  .ag-body-viewport,
  .ag-root-wrapper {
    scroll-behavior: smooth;
  }
  .ag-row .ag-cell {
    background-color: lighten($gms-prominent-background, 5%);
    border-left: 1px solid $gms-prominent-background;
    border-right: 1px solid $gms-prominent-background;
  }
  .ag-header-cell {
    padding: 0 10px;
    background-color: $gms-popover-background;
    .ag-header-cell-text {
      color: $gms-main;
    }
  }
  .ag-header-cell-label,
  .ag-header-cell-label {
    padding-left: 7px;
    font-size: 1rem;
    font-weight: 500;
  }
}
