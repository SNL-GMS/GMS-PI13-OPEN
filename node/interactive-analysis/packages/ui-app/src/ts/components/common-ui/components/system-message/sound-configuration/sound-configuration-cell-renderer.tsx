import { ColumnDefinition, DropDown, TableCellRenderer } from '@gms/ui-core-components';
import cloneDeep from 'lodash/cloneDeep';
import * as React from 'react';
import { userPreferences } from '~components/common-ui/config/user-preferences';
import { SoundConfigurationRendererParams, SoundConfigurationRow } from './types';

export type SoundConfigurationColumnDefinition = ColumnDefinition<
  SoundConfigurationRow,
  {},
  string | number,
  {},
  {}
>;

const defaultColumnWidthPx = 200;

const getColumnPosition = (props: SoundConfigurationRendererParams): number | 'first' | 'last' => {
  const index = props.columnApi
    .getAllDisplayedColumns()
    .findIndex(c => c.getColId() === props.colDef.colId);
  return index === 0
    ? 'first'
    : index === props.columnApi.getAllDisplayedColumns().length - 1
    ? 'last'
    : index;
};

export const NotificationsStatusCellRenderer: React.FunctionComponent<SoundConfigurationRendererParams> = props =>
  props.data?.hasNotificationStatusError ? (
    <TableCellRenderer
      className={`sound-configuration--notification-status-no-file-error`}
      data-col-position={getColumnPosition(props)}
      value={'!'}
      tooltipMsg={userPreferences.configuredAudibleNotificationFileNotFound(
        props.data.sound.selectedSound
      )}
      shouldCenterText={true}
    >
      {props.children}
    </TableCellRenderer>
  ) : null;

export const SoundConfigurationCellRenderer: React.FunctionComponent<SoundConfigurationRendererParams> = props => (
  <TableCellRenderer
    data-col-position={getColumnPosition(props)}
    value={props.valueFormatted ?? props.value}
  />
);

export const SoundConfigurationDropdownRenderer: React.FunctionComponent<SoundConfigurationRendererParams> = props => {
  if (!props.data || !props.data.sound) return null;

  const availableSounds = cloneDeep(props.data.sound.availableSounds);

  // if the configured sound happens to no longer be listed in the available sounds; add it
  // this will allow the selection to be correct and for an error to be displayed
  if (
    props.data.sound.selectedSound !== 'None' &&
    !availableSounds[props.data.sound.selectedSound]
  ) {
    availableSounds[props.data.sound.selectedSound] = props.data.sound.selectedSound;
  }
  return (
    <DropDown
      className={'sound-configuration__dropdown'}
      dropDownItems={availableSounds}
      value={props.data.sound.selectedSound}
      onMaybeValue={v => props.data.sound.onSelect(v)}
      title={'Select a sound'}
    />
  );
};

const headerCellBlockClass = 'soh-header-cell';

export const defaultColumnDefinition: SoundConfigurationColumnDefinition = {
  headerClass: `${headerCellBlockClass} ${headerCellBlockClass}--neutral`,
  width: defaultColumnWidthPx,
  sortable: true,
  filter: true,
  disableStaticMarkupForHeaderComponentFramework: true,
  disableStaticMarkupForCellRendererFramework: true,
  cellRendererFramework: SoundConfigurationCellRenderer
};
