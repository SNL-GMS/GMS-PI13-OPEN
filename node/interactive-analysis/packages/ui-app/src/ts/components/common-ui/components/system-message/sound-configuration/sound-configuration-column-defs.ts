import { ColumnDefinition } from '@gms/ui-core-components';
import {
  NotificationsStatusCellRenderer,
  SoundConfigurationDropdownRenderer
} from './sound-configuration-cell-renderer';
import { SoundConfigurationRow } from './types';

export const columnDefs: ColumnDefinition<SoundConfigurationRow, {}, {}, {}, {}>[] = [
  {
    headerName: '',
    field: 'hasNotificationStatusError',
    width: 40,
    minWidth: 40,
    resizable: false,
    suppressMovable: true,
    cellRendererFramework: NotificationsStatusCellRenderer,
    suppressFilter: true,
    suppressSorting: true
  },
  {
    headerName: 'Sound',
    field: 'sound',
    cellRendererFramework: SoundConfigurationDropdownRenderer
  },
  {
    headerName: 'Category',
    field: 'category'
  },
  {
    headerName: 'Subcategory',
    field: 'subcategory'
  },
  {
    headerName: 'Severity',
    field: 'severity'
  },
  {
    headerName: 'Message',
    field: 'message',
    flex: 1,
    minWidth: 300
  }
];
