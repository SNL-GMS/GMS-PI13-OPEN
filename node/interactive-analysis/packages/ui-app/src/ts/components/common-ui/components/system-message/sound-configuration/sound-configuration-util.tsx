import { UserProfileTypes } from '@gms/common-graphql';
import { AVAILABLE_SOUND_FILES } from '@gms/common-util';
import { UILogger } from '@gms/ui-apollo';
import { Toaster } from '@gms/ui-util';
import Axios from 'axios';
import includes from 'lodash/includes';
import sortBy from 'lodash/sortBy';
import uniq from 'lodash/uniq';
import { userPreferences } from '~components/common-ui/config/user-preferences';
import { ALL_CATEGORIES, ALL_SEVERITIES, ALL_SUBCATEGORIES, SoundConfigurationRow } from './types';

const BASE_SOUNDS_PATH = userPreferences.baseSoundsPath;

const toaster = new Toaster();

/**
 * Determine which sound files are missing, if any.
 * @returns the missing sound files; empty array if no files are missing
 */
export const getMissingSounds = async (filenames: string[]): Promise<string[]> =>
  Promise.all<string>(
    sortBy<string>(uniq<string>(filenames)).map(async soundFileName =>
      Axios.get(`${BASE_SOUNDS_PATH}${String(soundFileName)}`)
        .then(response => undefined)
        .catch(error => soundFileName)
    )
  ).then(response => response.filter(r => r !== undefined));

/**
 * Validate the available sound files. Toast an error message for
 * each sound file that is not loaded properly (not found).
 */
export const validateAvailableSounds = (): void => {
  getMissingSounds(AVAILABLE_SOUND_FILES)
    .then(missingSoundFiles => {
      missingSoundFiles.forEach(soundFileName => {
        toaster.toastError(userPreferences.soundFileNotFound(soundFileName));
        UILogger.Instance().error(
          userPreferences.soundFileNotFound(`${BASE_SOUNDS_PATH}${soundFileName}`)
        );
      });
    })
    .catch(error => UILogger.Instance().error(error));
};

/**
 * Validate the configured audible notifications against the available sounds.
 * Toast an error message for each sound file that is configured for an audible notifications but not available.
 */
export const validateConfiguredAudibleNotifications = (
  audibleNotifications: UserProfileTypes.AudibleNotification[]
): void => {
  const configuredSoundFiles = sortBy<string>(
    uniq<string>(audibleNotifications?.map(notification => notification.fileName))
  );

  getMissingSounds(configuredSoundFiles)
    .then(missingSoundFiles => {
      const soundFilesInError = configuredSoundFiles.filter(sound =>
        includes(missingSoundFiles, sound)
      );
      soundFilesInError.forEach(soundFileName => {
        toaster.toastError(
          userPreferences.configuredAudibleNotificationFileNotFound(soundFileName)
        );
        UILogger.Instance().error(
          userPreferences.configuredAudibleNotificationFileNotFound(
            `${BASE_SOUNDS_PATH}${soundFileName}`
          )
        );
      });
    })
    .catch(error => UILogger.Instance().error(error));
};

/**
 * @returns the filenames of the sound files that are in the /sounds directory
 * Relative path only. For example /sounds/example.mp3 would be returned as "example.mp3"
 */
export const getAvailableSounds = () => {
  const soundFileNames: string[] = AVAILABLE_SOUND_FILES;
  const obj: { [key: string]: string } = {};
  let soundsMap = soundFileNames.reduce((prev, curr) => {
    prev[curr] = curr;
    return prev;
  }, obj);
  soundsMap = { default: 'None', ...soundsMap };
  return soundsMap;
};

/**
 * Filter rows based on severity, category, and subcategory
 * @param selectedSeverity - selected severity
 * @param rows - The rows to be filtered
 * @param selectedCategory - Selected category
 * @param selectedSubcategory - Selected subcategory
 *
 * @returns - Filtered sound configuration rows
 */
export const filterRowsByType = (
  selectedSeverity: string,
  rows: SoundConfigurationRow[],
  selectedCategory: string,
  selectedSubcategory: string
) => {
  let newRows = [...rows];

  if (selectedSeverity !== ALL_SEVERITIES) {
    newRows = newRows.filter(row => row.severity === selectedSeverity);
  }
  if (selectedCategory !== ALL_CATEGORIES) {
    newRows = newRows.filter(row => row.category === selectedCategory);
  }
  if (selectedSubcategory !== ALL_SUBCATEGORIES) {
    newRows = newRows.filter(row => row.subcategory === selectedSubcategory);
  }
  return newRows;
};
