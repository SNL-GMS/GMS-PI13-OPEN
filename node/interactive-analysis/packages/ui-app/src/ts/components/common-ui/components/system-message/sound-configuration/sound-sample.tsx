import { UILogger } from '@gms/ui-apollo';
import React, { createRef, useEffect, useState } from 'react';

interface SoundSampleProps {
  soundToPlay: string;
}

export const SoundSample: React.FunctionComponent<SoundSampleProps> = (props: SoundSampleProps) => {
  const ref = createRef<HTMLAudioElement>();
  const filename: string = props.soundToPlay?.split('/').slice(-1)[0];

  const [play, setPlay] = useState(false);

  useEffect(() => {
    if (play && props.soundToPlay && ref.current && filename !== 'None') {
      ref.current.play().catch(e => {
        UILogger.Instance().error(`Error playing sound "${props.soundToPlay}": ${e}`);
      });
    }
    setPlay(true);
  }, [props.soundToPlay]);

  if (filename === 'None') {
    return null;
  }
  return <audio ref={ref} src={props.soundToPlay} autoPlay={false} />;
};
