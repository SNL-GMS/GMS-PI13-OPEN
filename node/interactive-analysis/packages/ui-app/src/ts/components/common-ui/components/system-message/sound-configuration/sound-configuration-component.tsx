import { Button, Card, Dialog } from '@blueprintjs/core';
import { SystemMessageTypes, UserProfileTypes } from '@gms/common-graphql';
import { UILogger } from '@gms/ui-apollo';
import { InfiniteTable, useDatasource } from '@gms/ui-core-components';
import classNames from 'classnames';
import includes from 'lodash/includes';
import React, { useState } from 'react';
import { userPreferences } from '~components/common-ui/config/user-preferences';
import {
  getHeaderHeight,
  getRowHeightWithBorder
} from '~components/data-acquisition-ui/shared/table/utils';
import { AudibleNotificationContext } from '../audio/audible-notification-context';
import { defaultColumnDefinition } from './sound-configuration-cell-renderer';
import { columnDefs } from './sound-configuration-column-defs';
import { SoundConfigurationToolbar } from './sound-configuration-toolbar';
import { filterRowsByType, getAvailableSounds, getMissingSounds } from './sound-configuration-util';
import { SoundSample } from './sound-sample';
import {
  ALL_CATEGORIES,
  ALL_SEVERITIES,
  ALL_SUBCATEGORIES,
  FILTER_TYPE,
  SelectedOptions as SelectedFilterOptions,
  SoundConfigurationRow
} from './types';

interface SoundConfigurationProps {
  isVisible: boolean;
  systemMessageDefinitions: SystemMessageTypes.SystemMessageDefinition[];
  onToggle(): void;
}

/**
 * Custom hook to manage which rows are filtered and which are not.
 * @param rows the rows to filter
 * @returns
 * an the filtered rows;
 * a callback function to call when a filter is changed;
 * and a SelectedFilterOptions object containing the state of each filter
 */
const useRowFilters = (
  rows: SoundConfigurationRow[]
): [SoundConfigurationRow[], (type: FILTER_TYPE, value: any) => void, SelectedFilterOptions] => {
  const [selectedSeverity, setSelectedSeverity] = useState(ALL_SEVERITIES);
  const [selectedCategory, setSelectedCategory] = useState(ALL_CATEGORIES);
  const [selectedSubcategory, setSelectedSubcategory] = useState(ALL_SUBCATEGORIES);

  const onFilterChanged = (type: FILTER_TYPE, value): void => {
    switch (type) {
      case FILTER_TYPE.SEVERITY:
        setSelectedSeverity(value);
        break;
      case FILTER_TYPE.CATEGORY:
        setSelectedCategory(value);
        break;
      case FILTER_TYPE.SUBCATEGORY:
        setSelectedSubcategory(value);
        break;
      default:
        // This should never happen...
        // tslint:disable-next-line: no-console
        console.error('Unknown SystemMessage category');
    }
  };

  const filteredRows = filterRowsByType(
    selectedSeverity,
    rows,
    selectedCategory,
    selectedSubcategory
  );
  return [
    filteredRows,
    onFilterChanged,
    {
      selectedSeverity,
      selectedCategory,
      selectedSubcategory
    }
  ];
};

/**
 * creates a list of sound configuration rows.
 * @param systemMessageDefinitions the defined system messages from which to generate rows
 * @param onSelect called when a selection is made.
 */
const useSoundConfigurationRows = (
  systemMessageDefinitions: SystemMessageTypes.SystemMessageDefinition[],
  tableRef: React.MutableRefObject<InfiniteTable<SoundConfigurationRow, {}>>,
  onSelect?: (fileName: string) => void
): SoundConfigurationRow[] => {
  const context = React.useContext(AudibleNotificationContext);

  const [missingSoundFiles, setMissingSoundFiles] = useState<string[]>(null);

  const configuredAudibleNotifications = context.audibleNotifications?.map(a => a.fileName);

  React.useEffect(() => {
    getMissingSounds(configuredAudibleNotifications)
      .then(response => {
        setMissingSoundFiles(response);
      })
      .catch(error => UILogger.Instance().error(error));
  }, [context]);

  const availableSounds = getAvailableSounds();

  const getSelectedSound = (
    systemMessageDefinition: SystemMessageTypes.SystemMessageDefinition
  ): string =>
    context.audibleNotifications?.find(
      notification =>
        SystemMessageTypes.SystemMessageType[notification.notificationType] ===
        SystemMessageTypes.SystemMessageType[systemMessageDefinition.systemMessageType]
    )?.fileName ?? 'None';

  const rows: SoundConfigurationRow[] = systemMessageDefinitions.map(
    (systemMessageDefinition, idx) => ({
      id: `${idx}`,
      sound: {
        availableSounds,
        selectedSound: getSelectedSound(systemMessageDefinition),
        onSelect: selectedFileName => {
          const notification: UserProfileTypes.AudibleNotification = {
            fileName: selectedFileName === 'None' ? '' : selectedFileName,
            notificationType:
              SystemMessageTypes.SystemMessageType[systemMessageDefinition.systemMessageType]
          };

          context
            .setAudibleNotifications([notification])
            .then(result => {
              onSelect(selectedFileName);
              const tableApi = tableRef.current.getTableApi();
              tableApi.refreshInfiniteCache();
            })
            .catch(err => {
              UILogger.Instance().error(err);
            });
        }
      },
      hasNotificationStatusError:
        missingSoundFiles &&
        missingSoundFiles.length > 0 &&
        includes(missingSoundFiles, getSelectedSound(systemMessageDefinition)),
      category: systemMessageDefinition.systemMessageCategory,
      subcategory: systemMessageDefinition.systemMessageSubCategory,
      severity: systemMessageDefinition.systemMessageSeverity,
      message: systemMessageDefinition.template
    })
  );

  return rows;
};

/**
 * The actual component that creates a dialog, chooses and plays sounds.
 * Calls on the context.setAudibleNotifications to update the user profile as
 * soon as user selects a sound.
 */
const SoundConfigurationComponent: React.FunctionComponent<SoundConfigurationProps> = ({
  isVisible,
  onToggle,
  systemMessageDefinitions
}) => {
  const [sampleSound, setSampleSound] = useState<string>(null);
  const tableRef = React.useRef<InfiniteTable<SoundConfigurationRow, {}>>(null);
  const setSampleSoundWithPath = (s: string) =>
    setSampleSound(`${userPreferences.baseSoundsPath}${s}`);
  const rows = useSoundConfigurationRows(
    systemMessageDefinitions,
    tableRef,
    setSampleSoundWithPath
  );
  const [filteredRows, onFilterChanged, selectedOptions] = useRowFilters(rows);
  const datasource = useDatasource(filteredRows);

  return (
    <Dialog
      onClose={onToggle}
      isOpen={isVisible}
      autoFocus={true}
      hasBackdrop={true}
      canEscapeKeyClose={true}
      title={'Sound Configuration'}
      style={{ width: 'auto' }}
      canOutsideClickClose={true}
    >
      <div className={classNames('sound-configuration ag-theme-dark')}>
        <SoundSample soundToPlay={sampleSound} />
        <Card interactive={false}>
          <SoundConfigurationToolbar
            systemMessageDefinitions={systemMessageDefinitions}
            selectedOptions={selectedOptions}
            onChanged={onFilterChanged}
          />
          <div className={'sound-configuration__container'}>
            <InfiniteTable<SoundConfigurationRow, {}>
              ref={ref => (tableRef.current = ref)}
              id="table-sound-configuration"
              key="table-sound-configuration"
              datasource={datasource}
              context={{}}
              defaultColDef={defaultColumnDefinition}
              columnDefs={columnDefs}
              rowData={filteredRows}
              rowHeight={getRowHeightWithBorder()}
              headerHeight={getHeaderHeight()}
              getRowNodeId={node => node.id}
              deltaRowDataMode={true}
              rowDeselection={true}
              overlayNoRowsTemplate="No sounds to display"
              suppressContextMenu={true}
            />
          </div>
          <div className={'sound-configuration__footer'}>
            <Button className={'sound-configuration__close-button'} onClick={onToggle}>
              Close
            </Button>
          </div>
        </Card>
      </div>
    </Dialog>
  );
};

/**
 * Dialog sound configuration component. The purpose of this component is to give
 * the user the ability to select a sound from a predefined list for specific
 * events that happen.
 */
export const SoundConfiguration = React.memo(SoundConfigurationComponent);
