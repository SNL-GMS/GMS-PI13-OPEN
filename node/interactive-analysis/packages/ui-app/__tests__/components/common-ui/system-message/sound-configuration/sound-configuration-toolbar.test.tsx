import * as React from 'react';

import { SoundConfigurationToolbar } from '../../../../../src/ts/components/common-ui/components/system-message/sound-configuration/sound-configuration-toolbar';
import {
  ALL_CATEGORIES,
  ALL_SEVERITIES,
  ALL_SUBCATEGORIES
} from '../../../../../src/ts/components/common-ui/components/system-message/sound-configuration/types';
import { systemMessageDefinitions } from '../../../../__data__/common-ui/system-message-definition-data';
// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe('Sound Configuration Toolbar', () => {
  const soundConfiguration = Enzyme.mount(
    <SoundConfigurationToolbar
      systemMessageDefinitions={systemMessageDefinitions}
      selectedOptions={{
        selectedSeverity: ALL_SEVERITIES,
        selectedCategory: ALL_CATEGORIES,
        selectedSubcategory: ALL_SUBCATEGORIES
      }}
      onChanged={jest.fn()}
    />
  );

  it('should be defined', () => {
    expect(SoundConfigurationToolbar).toBeDefined();
    expect(soundConfiguration).toBeDefined();
  });

  it('matches snapshot', () => {
    soundConfiguration.update();
    expect(soundConfiguration).toMatchSnapshot();
  });
});
