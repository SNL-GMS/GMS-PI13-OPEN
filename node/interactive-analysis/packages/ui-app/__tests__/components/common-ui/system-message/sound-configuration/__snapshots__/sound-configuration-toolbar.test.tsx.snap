// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sound Configuration Toolbar matches snapshot 1`] = `
<Component
  onChanged={[MockFunction]}
  selectedOptions={
    Object {
      "selectedCategory": "All categories",
      "selectedSeverity": "All severities",
      "selectedSubcategory": "All subcategories",
    }
  }
  systemMessageDefinitions={
    Array [
      Object {
        "systemMessageCategory": "SOH",
        "systemMessageSeverity": "INFO",
        "systemMessageSubCategory": "USER",
        "systemMessageType": "CHANNEL_MONITOR_TYPE_QUIET_PERIOD_CANCELED",
        "template": "Station %s Channel %s %s quiet period canceled by user %s",
      },
      Object {
        "systemMessageCategory": "SOH",
        "systemMessageSeverity": "INFO",
        "systemMessageSubCategory": "USER",
        "systemMessageType": "CHANNEL_MONITOR_TYPE_STATUS_CHANGE_ACKNOWLEDGED",
        "template": "Station %s Channel %s %s status change acknowledged by user %s|Station %s Channel %s %s status change acknowledged by user %s with comment '%s'",
      },
    ]
  }
>
  <div
    className="bp3-html-select sound-configuration-toolbar"
  >
    <div
      className="sound-configuration-toolbar__dropdown"
    >
      <DropDown
        dropDownItems={
          Object {
            "INFO": "INFO",
            "default": "All severities",
          }
        }
        onMaybeValue={[Function]}
        title="Filter by severity"
        value="All severities"
      >
        <HTMLSelect
          onChange={[Function]}
          title="Filter by severity"
          value="All severities"
        >
          <div
            className="bp3-html-select"
          >
            <select
              multiple={false}
              onChange={[Function]}
              title="Filter by severity"
              value="All severities"
            >
              <option
                key="INFO"
                value="INFO"
              >
                INFO
              </option>
              <option
                key="default"
                value="All severities"
              >
                All severities
              </option>
            </select>
            <Blueprint3.Icon
              icon="double-caret-vertical"
            >
              <span
                className="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </Blueprint3.Icon>
          </div>
        </HTMLSelect>
      </DropDown>
    </div>
    <div
      className="sound-configuration-toolbar__dropdown"
    >
      <DropDown
        dropDownItems={
          Object {
            "SOH": "SOH",
            "default": "All categories",
          }
        }
        onMaybeValue={[Function]}
        title="Filter by category"
        value="All categories"
      >
        <HTMLSelect
          onChange={[Function]}
          title="Filter by category"
          value="All categories"
        >
          <div
            className="bp3-html-select"
          >
            <select
              multiple={false}
              onChange={[Function]}
              title="Filter by category"
              value="All categories"
            >
              <option
                key="SOH"
                value="SOH"
              >
                SOH
              </option>
              <option
                key="default"
                value="All categories"
              >
                All categories
              </option>
            </select>
            <Blueprint3.Icon
              icon="double-caret-vertical"
            >
              <span
                className="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </Blueprint3.Icon>
          </div>
        </HTMLSelect>
      </DropDown>
    </div>
    <div
      className="sound-configuration-toolbar__dropdown"
    >
      <DropDown
        dropDownItems={
          Object {
            "USER": "USER",
            "default": "All subcategories",
          }
        }
        onMaybeValue={[Function]}
        title="Filter by subcategory"
        value="All subcategories"
      >
        <HTMLSelect
          onChange={[Function]}
          title="Filter by subcategory"
          value="All subcategories"
        >
          <div
            className="bp3-html-select"
          >
            <select
              multiple={false}
              onChange={[Function]}
              title="Filter by subcategory"
              value="All subcategories"
            >
              <option
                key="USER"
                value="USER"
              >
                USER
              </option>
              <option
                key="default"
                value="All subcategories"
              >
                All subcategories
              </option>
            </select>
            <Blueprint3.Icon
              icon="double-caret-vertical"
            >
              <span
                className="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </Blueprint3.Icon>
          </div>
        </HTMLSelect>
      </DropDown>
    </div>
  </div>
</Component>
`;
